backend/fbdev.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Twin - A Tiny Window System
+ * Copyright (c) 2024 National Cheng Kung University, Taiwan
+ * All rights reserved.
+ */
+
+#include <fcntl.h>
+#include <linux/fb.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/ioctl.h>
+#include <sys/mman.h>
+#include <twin.h>
+#include <unistd.h>
+
+#include "twin_backend.h"
+
+typedef struct {
+    int fd;
+    int depth;
+    int image_y;
+    int *pixels;
+    int *fbdev_raw;
+    twin_screen_t *screen;
+    twin_coord_t width, height;
+} twin_fbdev_t;
+
+#define FBDEV_ENVVAR "FRAMEBUFFER"
+#define SCREEN(x) ((twin_context_t *) x)->screen
+#define PRIV(x) ((twin_fbdev_t *) ((twin_context_t *) x)->priv)
+
+static void _twin_fbdev_put_begin(twin_coord_t left,
+                                  twin_coord_t top,
+                                  twin_coord_t right,
+                                  twin_coord_t bottom,
+                                  void *closure)
+{
+    twin_fbdev_t *tx = PRIV(closure);
+    tx->width = right - left;
+    tx->height = bottom - top;
+    tx->image_y = top;
+}
+
+static void _twin_fbdev_put_span(twin_coord_t left,
```

Comment:
Consider the reworked:
```c
static void _twin_fbdev_put_span (twin_coord_t    left,
                                  twin_coord_t    top,
                                  twin_coord_t    right,
                                  twin_argb32_t   *pixels,
                                  void            *closure)
{
    twin_screen_t *screen = SCREEN(closure);
    twin_fbdev_t *tx = PRIV(closure);
    ...
    memcpy(dest, pixels, width * sizeof(unsigned int));
}
```

Reply:
What I understand here is that we just update the whole Linux framebuffer whenever a sub-region needs to be updated (i.e., calling `_twin_fbdev_put_span()`).